pkg/cmd/issue/comment/comment.go

```diff
@@ -0,0 +1,215 @@
+		`),
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			// support `-R, --repo` override
+			opts.BaseRepo = f.BaseRepo
+			opts.SelectorArg = args[0]
+			opts.Interactive = !(opts.Body != "" || webMode || editorMode)
+
+			if !opts.IO.CanPrompt() && opts.Interactive {
+				return &cmdutil.FlagError{Err: errors.New("--body, --editor, or --web required when not running interactively")}
+			}
+
+			if !opts.Interactive {
+				if opts.Body != "" && !webMode && !editorMode {
+					opts.InputType = inline
+				} else if opts.Body == "" && webMode && !editorMode {
+					opts.InputType = web
+				} else if opts.Body == "" && !webMode && editorMode {
+					opts.InputType = editor
+				} else {
+					return &cmdutil.FlagError{Err: fmt.Errorf("specify only one of --body, --editor, or --web")}
+				}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+			return commentRun(opts)
+		},
+	}
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "", "Supply a body. Will prompt for one otherwise.")
+	cmd.Flags().BoolVarP(&editorMode, "editor", "e", false, "Add body using editor")
+	cmd.Flags().BoolVarP(&webMode, "web", "w", false, "Add body in browser")
+
+	return cmd
+}
+
+func commentRun(opts *CommentOptions) error {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	apiClient := api.NewClientFromHTTP(httpClient)
+
+	issue, baseRepo, err := shared.IssueFromArg(apiClient, opts.BaseRepo, opts.SelectorArg)
+	if err != nil {
+		return err
+	}
+
+	if opts.Interactive {
+		inputType, err := inputTypeSurvey()
+		if err != nil {
+			return err
+		}
+		opts.InputType = inputType
+	}
+
+	switch opts.InputType {
+	case web:
+		openURL := issue.URL + "#issuecomment-new"
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.ErrOut, "Opening %s in your browser.\n", utils.DisplayURL(openURL))
+		}
+		return utils.OpenInBrowser(openURL)
+	case inline:
+		if opts.Body != "" {
+			break
+		}
+		body, err := inlineSurvey()
+		if err != nil {
+			return err
+		}
+		opts.Body = body
+	case editor:
+		editorCommand, err := cmdutil.DetermineEditor(opts.Config)
+		if err != nil {
+			return err
+		}
+		var body string
+		if opts.Interactive {
+			body, err = editorSurvey(editorCommand)
+		} else {
+			body, err = opts.Edit(editorCommand)
+		}
+		if err != nil {
+			return err
+		}
+		opts.Body = body
+	}
+
+	if opts.Interactive {
+		cont, err := confirmSubmitSurvey()
+		if err != nil {
+			return err
+		}
+		if !cont {
+			return fmt.Errorf("Discarding...")
+		}
+	}
+
+	params := map[string]string{
+		"subjectId": issue.ID,
+		"body":      opts.Body,
+	}
+
+	url, err := api.CommentCreate(apiClient, baseRepo.RepoHost(), params)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprint(opts.IO.Out, url)
+
+	return nil
+}
+
+func inputTypeSurvey() (int, error) {
+	var inputType int
+	inputTypeQuestion := &survey.Select{
+		Message: "Where do you want to draft your comment?",
+		Options: []string{"In line", "Editor", "Web"},
```

Comment:
I'm not sure if this survey prompt is useful. I'm imagining that, if our users were to write a comment interactively, they would appreciate if we launched a text editor immediately.

Would we consider `--editor` being implied (and removing it as an explicit flag) if `--body` and `--web` were not supplied? And then launching an editor directly without prompting, since the user already gave us an indication that they want to write a comment?

That would mean that we would only interface with the Survey library to show the `Submit?` step once the editor finishes.

Reply:
I like this optimization, it makes this command work similar to `git commit`. I wonder if we would even want a `Submit?` prompt in that case. @ampinsk @vilmibm do you have any thoughts?

Reply:
I think jumping straight into the editor from the prompt would be jarring, especially because it's very unlike the rest of gh to do so. I'd prefer to keep the design of this as is, but if you both feel strongly I'm open to discussing it! Curious if @vilmibm has thoughts.

Reply:
I'm unsure about this and for that reason do want to ship it like this; i want to see how it feels and if we get feedback on it.

i think we're still establishing best practices around interactive modes and some inconsistency / extra verbosity might help us narrow down those practices.

Reply:
> I think jumping straight into the editor from the prompt would be jarring, especially because it's very unlike the rest of gh to do so.

Agreed that it might be jarring. But for a purpose like editing, where there is only one field to collect (as opposed to opening an issue or PR), I think that it could be something that we could get away with. However, if we were to also provide the `Web` option, then absolutely, the first prompt is in order.

My objection was mostly for providing the `In line` vs. `Editor` option, as I don't understand why users would choose the `In line` option (where neither readline nor terminal escape sequences are supported), over their full-featured editor?

The second thing that I believe can reduce friction is that we skip the additional `[e] to launch editor` prompt which follows the user's choice of `Editor`, and jump straight into editing instead:
```
? Where do you want to draft your comment? Editor
? Body [(e) to launch vim]     <--- does anyone benefit from this?
```

Reply:
I played around with this a bit and decided to keep the input survey but removed the `In line` option. The main reason was because `survey` does not use readline and does not handle terminal escape sequences well, this made the overall editing experience poor. If users want to supply the body message in line they will need to use the `--body` flag. 

I agree that the second prompt to open the editor is unnecessary and will see if I can remove it.